### ISSUE TITLE
Oxyplot Bad Performance

### ISSUE THREAD

**Issue by Nick135** [oxyplot/oxyplot#1865]
I realy like oxyplot but the performance with large line series is terrible.
Currently I use the WPF model of an old Version of the lib inside a WinForm Projekt.

I make a view test with the oxplot demo (Current Version) and on my tests WinForm ist the fastest, then SkiaSharp and the slowest is WPF.
But the curson on WPF looks much more butyfull then the WinForms one.

Is there any interrest to make oxyplot really fast?
There is "oxyplot-sharpdx" but it looks like dead.

I don't wanna change to another lib.
What can I do to increase the performance of this lib?

**Comment by quinmars** [comment]
I think it is hard to talk about performance, without concrete numbers. We switch from the WPf backend to the WPF skia backend recently. I find the skia backend quite snappy. It smoothly shows ~200 000 data points. Well, I also got a new PC in that time, so I have not a fair comparison. How large are your line series?

**Comment by Nick135** [comment]
The Test with WinForm, WPF and WPF Skia I only made with the Oxyplot Demo -> Performance -> LineSeries, 1M points.

My app:
For example, when I have a trace with 10 signals and 2ms Sampling rate I got 1.8M points/series (18M points) per hour.

Default is a sampling rate from 4-10ms but 20 signals. So I got ~450k / series

```
[Example("LineSeries, 1M points - 10 Series")]
public static PlotModel LineSeries1M10_Series()
{
    var model = new PlotModel { Title = "LineSeries, 1M points - 10 Series" };
    for (int i = 0; i < 10; i++)
    {
        var s = new LineSeries();
        AddPoints(s.Points, 1000000);
        model.Series.Add(s);
    }
    return model;
}
````

**Comment by Nick135** [comment]
On my gaming pc the perfermance of WPF skia is not so bad on move and zoom. Similar as winforms.
But the cursor is much faster with skia.

When I look in my task manager or grafikkarte software only one cpu core do all the work.
GPU is on 0-2% for all examples.

**Comment by Nick135** [comment]
Is there any benefit on using netcoreapp3.1 instead of full Framework? Now there is only a netcoreapp3.1 Version.

My App currently based on .Net 4.0 and can run on all Windows since WinXP and on the most PCs without installation.
With .Net Core I need administration rights on the PC, what the normal user in our company don´t have.

**Comment by VisualMelon** [comment]
All existing releases target .NET 4.0 (though Skia may not work, I'm not sure). I'd be surprise if .NET Core perf was worse, and it has certainly become better over the years.

Obviously I don't know the policy at your company specifically, but generally you don't need any special privileges for .NET Core or .NET 5/6 more than for other applications: are you able to try publishing your application as a single-file application?

**Comment by Nick135** [comment]
I don't need any special privileges for .Net Core, but there must be .Net Core installed and this isn't on a default Windows installation.

**Comment by Nick135** [comment]
I made some tests with newer dot net console apps in the past and the performance decrease with the number of outputs.
With output .Net 4.0 is faster then 4.x and more faster then .Net Core.

**Comment by Nick135** [comment]
Is there any option to increase the performance by using the GPU ore use multi threading?
Mobile CPUs are not that powerful in single threading.

**Comment by VisualMelon** [comment]
No, there's no threading/GPU support; the backend is very simple really.

> I don't need any special privileges for .Net Core, but there must be .Net Core installed and this isn't on a default Windows installation.

You don't need it installed if you package it as a single-file application: see https://docs.microsoft.com/en-us/dotnet/core/deploying/single-file/overview
